static/resources/harness.js

```diff
@@ -144,59 +177,62 @@
   //
   // If the test doesn't return true or false, or if it throws, `result` will
   // be null and a `message` property is set to an explanation.
-  function test(data) {
+  function test(data, callback) {
     var result = {name: data.name, info: {}};
 
-    for (var i = 0; i < data.tests.length; i++) {
+    function runTest(i, callback) {
+      function process(value) {
+        processValue(value, data, test, result);
```

Comment:
Some comments to explain what's going on here would be great. How does a mix of sync and async tests work? Given only async tests, are they run one after the other or all in parallel? If they're run in series, what happens between one test finishing and the next one starting? Is there a `setTimeout` somewhere, or is there no async step between two tests?